### PR TITLE
Fixed IDE services touching `notnull` constraint

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Looking good.  Did i miss it, or are there no completion tests here?

**Comment by AlekseyTs** [comment]
> Did i miss it, or are there no completion tests here?

NotNullKeywordRecommenderTests were added in https://github.com/dotnet/roslyn/pull/36109. Is this what you were looking for?

**Comment by CyrusNajmabadi** [comment]
Indeed.  Sorry, i missed that PR the last time around!

File: src/Compilers/VisualBasic/Portable/Symbols/TypeParameterSymbol.vb [hunk 1]
```diff
@@ -320,6 +320,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             End Get
         End Property
 
+        Private ReadOnly Property HasNotNullConstraint As Boolean Implements ITypeParameterSymbol.HasNotNullConstraint
```
Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/Symbols/TypeParameterSymbol.vb]: >HasNotNullConstraint [](start = 34, length = 20)

Consider adding an API test for VB (we already had tests for the C# API)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-ide Please review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-ide Please review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-ide Please review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-ide Please review

File: src/Features/CSharp/Portable/Completion/CompletionProviders/KeywordCompletionProvider.cs [hunk 2]
```diff
@@ -97,6 +97,7 @@ private static ImmutableArray<IKeywordRecommender<CSharpSyntaxContext>> GetKeywo
                 new NameOfKeywordRecommender(),
                 new NamespaceKeywordRecommender(),
                 new NewKeywordRecommender(),
+                new NotNullKeywordRecommender(),
```
Comment (genlu) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/KeywordCompletionProvider.cs]: Is there a test for this?

Comment (AlekseyTs) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/KeywordCompletionProvider.cs]: > Is there a test for this?

What kind of test do you  have in mind?

Comment (genlu) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/KeywordCompletionProvider.cs]: ~~Check if `notnull` shows as keyword in completion list~~

Ah, I see I have missed the comments above. Please ignore this

**Pull request merged**